tests/test_indexing.py

```diff
@@ -0,0 +1,71 @@
+from historydag.dag import history_dag_from_newicks
```

Comment:
I think what we should do is move these tests to `test_factory.py`. That file contains the variables `dags` and `cdags`, which are both lists of history DAGs, including some bigger ones. Hopefully it won't take too long to run your tests on all of them?